### PR TITLE
Added Column NoSQL family support + Cassandra Support

### PR DESCRIPTION
**Pull request by dearrudam**
## Changes
- Added Column NoSQL family support
- Added Cassandra support
- Added info about this new support into the README.md;

### CODE REVIEW

**Comment by dearrudam** [comment]
@amoscatelli could you review this PR, please?!

I'd like to congrats you and the Quarkus team that is maintaining this amazing extension!!!

**Comment by amoscatelli** [comment]
@dearrudam thank you, gonna review now

File: column-parent/cassandra-parent/integration-tests/src/test/resources/init_script.cql
Comment (amoscatelli) [inline column-parent/cassandra-parent/integration-tests/src/test/resources/init_script.cql:1]: is this required ?
where is this used ?

File: column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java
Comment (amoscatelli) [inline column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java:1]: why is this required instead of using DefaultCassandraColumnManager directly ?

**Review by amoscatelli — COMMENTED** [review]
@dearrudam here you are

File: column-parent/cassandra-parent/deployment/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        <groupId>io.quarkiverse.jnosql</groupId>
+        <artifactId>quarkus-jnosql-column-cassandra-parent</artifactId>
+        <version>1.0.4-SNAPSHOT</version>
+    </parent>
+    <artifactId>quarkus-jnosql-column-cassandra-deployment</artifactId>
+    <name>Quarkus JNoSQL - Column - Cassandra - Deployment</name>
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jsonb-deployment</artifactId>
```
Comment (amoscatelli) [inline column-parent/cassandra-parent/deployment/pom.xml:14]: also, are you sure Cassandra JNoSQL driver requires Jsonb ?
That's true for Redis and DynamoDB

File: column-parent/cassandra-parent/runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <name>Quarkus JNoSQL - Column - Cassandra - Runtime</name>
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkiverse.jnosql</groupId>
+            <artifactId>quarkus-jnosql-column</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jsonb</artifactId>
```
Comment (amoscatelli) [inline column-parent/cassandra-parent/runtime/pom.xml:19]: the same applies here

**Review by amoscatelli — COMMENTED** [review]
please double check the jsonb dependency also

File: column-parent/cassandra-parent/integration-tests/src/test/resources/init_script.cql
Comment (dearrudam) [reply column-parent/cassandra-parent/integration-tests/src/test/resources/init_script.cql:1]: Cassandra requires KEYSPACEs, COMLUMNFAMILYs, or TABLEs defined to be able to be consumed for their clients. 

It's required in any scenario that we're getting started from scratch with a Cassandra database.

This file is used by the cassandra-quarkus-client extension to setup the Cassandra database on application initialization;

File: column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java
Comment (dearrudam) [reply column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java:1]: The DefaultCassandraColumnManager's close method closes the CqlSession instance. I think that it's not the behavior that we want for this component, once the CqlSession is provided by the cassandra-quarkus-client extension.

File: column-parent/cassandra-parent/runtime/pom.xml (hunk 2 above)
Comment (dearrudam) [reply column-parent/cassandra-parent/runtime/pom.xml]: I've removed and pushed up the change already.

File: column-parent/cassandra-parent/deployment/pom.xml (hunk 1 above)
Comment (dearrudam) [reply column-parent/cassandra-parent/deployment/pom.xml]: Yeah! It looks like it's not necessary! I've removed and pushed up the change already.

File: column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java
Comment (amoscatelli) [reply column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java:1]: What you say is generally true but, I think this may not be relevant in a microservice scenario.
When DefaultCassandraColumnManager is going to be shutted down, in a Quarkus application, is the very moment the whole cassandra-quarkus-client extension is going to be shutted down too (usually when scaling down, when killing a kubernetes pod).

We want our integration to stay as thin as possible in respect of the original JNoSQL so we don't have to do nothing more in case of a JNoSQL refactor.

Also, often, multiple close calls on the same istance should be ok (the second one simply does nothing).

So in short : 
Have you tested multiple close calls on the same CqlSession ? Does this lead to an error ?
Have you run your integration tests using the original DefaultCassandraColumnManager ?
Have you encountered any problem during the shutting down of a pod using the original DefaultCassandraColumnManager ?
Since QuarkusCassandraColumnManagerFactory already doesn't propagate close calls on the CqlSession object, are you sure the DefaultCassandraColumnManager close method is indeed called by the JNoSQL framework ?

To tell the truth I am just thinking out lout, because for other extensions I encountered no problems so far.
So I am just investigating this is really necessary.

If you confirm this is really necessary, I am going to merge right away.

Thank you for your patience.

Comment (amoscatelli) [reply column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java:1]: @dearrudam I decided to mrge anyway
Please let me know what you think of what I wrote, the same

Comment (dearrudam) [reply column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java:1]: >Have you tested multiple close calls on the same CqlSession ? Does this lead to an error ?

I didn't execute tests like stress tests against this implementation yet.

>Have you run your integration tests using the original DefaultCassandraColumnManager ?

No, I didn't.

>Have you encountered any problem during the shutting down of a pod using the original DefaultCassandraColumnManager ?

I didn't execute tests about the `DefaultCassandraColumnManager` implementation on a pod or any container environment yet, then I don't know about that for now.

> Since QuarkusCassandraColumnManagerFactory already doesn't propagate close calls on the CqlSession object, are you sure the DefaultCassandraColumnManager close method is indeed called by the JNoSQL framework ?

No, it's not called by the JNoSQL framework itself. But the close method is exposed in the ColumnManager interface then the clients may call it whenever they want. It's true for other manager interfaces too, like `KeyValueManager`, `ColumnManage`, `DocumentManager`, and so on because they extends `AutoCloseable` interface as well.

> I hope I explained my point of view, I know I making a fuss out of nothing :)

No worries! It makes sense to think about that! Maybe I've worried to avoid an undesired behavior prematurely... maybe no one will call the "close()" method during the whole application runtime lifecycle...  

Let me know if you prefer to change to use the DefaultCassandraColumnManager, okay? 

> p.s. also why didn't you add yourself in the All Contributors section ?

:laughing: I thought that it was an automated process that collects the collaborators and inserts them into the `.all-contributorsrc` file...

Comment (amoscatelli) [reply column-parent/cassandra-parent/runtime/src/main/java/org/eclipse/jnosql/databases/cassandra/communication/QuarkusCassandraColumnManager.java:1]: My bad, I was not clear.
I was talking about contributors list in the README file, I'll add you myself.
Yea the .all-contributorsrc is automatic ;)

Gonna make some test with DefaultCassandraColumnManager myself, if something bad happens we'll reintroduce the QuarkusCassandraColumnManager.

Even when using JPA clients can close/manipulate the Transaction using the EntityManager, even when both are instantiated by the application server, as already said I wouldn't pay too much attention atm.

Again thank you for your contribution !

If you know somebody that could be interested in contributing for the other drivers let me know so we can make a plan.

Maybe I'll start working on ArangoDB's ...

**Comment by dearrudam** [comment]
I have a question: when this feature will be available in the Maven Central repo?!

**Comment by amoscatelli** [comment]
I would like add at least another supported driver before releasing 1.0.4.
Can you wait a couple of days ?

**Comment by dearrudam** [comment]
That's okay!!! Do you want help with adding ArangoDB support?

**Comment by dearrudam** [comment]
I have another question: Quarkus have Devservices that start-up containers like MongoDB and Redis... do you know how to add this capability for other containers? I mean, it will help many developers to use the quarkus-jnosql extensions because most drivers require at least, in dev mode, their containers running...

**Comment by amoscatelli** [comment]
If you want to contribute with ArangoDB I can take another one.
I want to cover more and more of original JNoSQL drivers.

Regarding your question : that's a good question :-)
Dunno if we can use directly dev-containers ...

**Comment by amoscatelli** [comment]
Or maybe we can contribute and add missing databases to devservices, maybe it's not so hard

**Comment by dearrudam** [comment]
Yeah! Anyway, I think that we could focus on adding support for more drivers meanwhile we try to figure out how to contribute with adding databases to the Quarkus devservices...

**Comment by amoscatelli** [comment]
This is the pom.xml of the integration test for orangodb quarkus extension : 

https://github.com/apache/camel-quarkus/blob/main/integration-tests/arangodb/pom.xml

It seems I was right, they use testcontainers directly but I still need to study this

**Comment by dearrudam** [comment]
Yeah... I need to study this too... I think that it will improve the dev experience by providing this capability...
